File: Forasoft/JavaVersion/app/src/main/java/com/pustovit/forajava/view/AlbumActivity.java
package com.pustovit.forajava.view;

import android.os.Bundle;


import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.pustovit.forajava.R;
import com.pustovit.forajava.databinding.ContentFullAlbumBinding;
import com.pustovit.forajava.repository.model.Album;
import com.pustovit.forajava.repository.model.Song;
import com.pustovit.forajava.view.adapter.SongAdapter;

import java.util.List;
/**
 * Album detail Activity.
 * <p>
 * Created by Pustovit Vladimir on 28.01.2020.
 * vovapust1989@gmail.com
 */
public class AlbumActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_full_album);
        activateToolbar(true);
        setTitle("");

        ContentFullAlbumBinding contentBinding = DataBindingUtil.setContentView(AlbumActivity.this, R.layout.content_full_album);

        Bundle bundle = getIntent().getExtras();
        if (bundle != null) {
            Album album = bundle.getParcelable(Album.class.getName());
            List<Song> songList = bundle.getParcelableArrayList(Song.class.getName());
            contentBinding.setAlbum(album);
            SongAdapter songAdapter = new SongAdapter(songList);
            RecyclerView rvSongs = contentBinding.rvSongs;
            rvSongs.setLayoutManager(new LinearLayoutManager(this));
            rvSongs.setAdapter(songAdapter);
        }

    }

    void activateToolbar(boolean enableHome) {

        ActionBar actionBar = getSupportActionBar();
        if (actionBar == null) {
            Toolbar toolbar = findViewById(R.id.toolbar);

            if (toolbar != null) {
                setSupportActionBar(toolbar);
                actionBar = getSupportActionBar();
            }
        }

        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(enableHome);
        }
    }

}

File: Forasoft/KotlinVersion/app/src/main/java/com/pustovit/kotlinversion/view/AlbumActivity.kt
package com.pustovit.kotlinversion.view

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.pustovit.kotlinversion.R
import com.pustovit.kotlinversion.databinding.ContentAlbumBinding
import com.pustovit.kotlinversion.repository.model.Album
import com.pustovit.kotlinversion.repository.model.Song
import com.pustovit.kotlinversion.view.adapter.SongAdapter
import java.io.Serializable

class AlbumActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_album)
        activateToolbar(true)
        title = ""

        val contentBinding: ContentAlbumBinding =
            DataBindingUtil.setContentView(this@AlbumActivity, R.layout.content_album)

        val bundle = intent.extras
        if (bundle != null) {
            val mAlbum: Album = bundle.getSerializable(Album::class.java.getName()) as Album
            val mSongList: List<Song> = bundle.getSerializable(Song::class.java.getName())as  List<Song>

            contentBinding.setAlbum(mAlbum)

            val songAdapter = SongAdapter(mSongList)
            val rvSongs: RecyclerView = contentBinding.rvSongs
            rvSongs.layoutManager = LinearLayoutManager(this)
            rvSongs.adapter = songAdapter
        }

    }

    fun activateToolbar(enableHome: Boolean) {
        var actionBar = supportActionBar
        if (actionBar == null) {
            val toolbar =
                findViewById<Toolbar>(R.id.toolbar)
            if (toolbar != null) {
                setSupportActionBar(toolbar)
                actionBar = supportActionBar
            }
        }
        actionBar?.setDisplayHomeAsUpEnabled(enableHome)
    }
}
